lib/_http_client.js

```diff
@@ -270,6 +272,17 @@ function ClientRequest(input, options, cb) {
 
     if (host && !this.getHeader('host') && setHost) {
       let hostHeader = host;
+      let hostHeaderPort = port;
+
+      // Prevent potential domain fronting misjudgement when using proxy
+      // Overwrite the default Host request header with the host of path
+      if (!StringPrototypeStartsWith(this.path, '/')) {
+        const proxyURL = new URL(StringPrototypeMatch(
+          this.path, '^.+://') ? this.path : `http://${this.path}`);
```

Comment:
`RegExpPrototypeTest` is way more efficient than `StringPrototypeMatch` IIRC. It might be worth to store the RegEx in global variable to avoid re-building it at each call.
```suggestion
        const proxyURL = new URL(RegExpPrototypeTest(/^[^:]+:\/\//, this.path) ?
                                 this.path : `http://${this.path}`);
```